Added new test case of NVMe long-term stress test

juhlee-microsoft opened the pull request:
A simple twick of NVMe performance test case to Stress one. Measured multiple times of the testing time. The key parameter would be 'FIO_STRESS_RUNTIME'. which users can control the execution time.

TEST RESULT with'FIO_STRESS_RUNTIME=720'

[LISAv2 Test Results Summary]
Test Run On           : 11/06/2019 02:22:29
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Initial Kernel Version: 5.0.0-1023-azure
Final Kernel Version  : 5.0.0-1023-azure
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:7:30

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STRESS               STRESSTEST-NVME-4K-IO                                                             PASS               443.64
        Mode=randread : block_size=4K q_depth=64 iops=998403.720554
        Mode=randread : block_size=4K q_depth=128 iops=1602833.672088
        Mode=randread : block_size=4K q_depth=256 iops=2347870.182987
        Mode=randread : block_size=4K q_depth=512 iops=2837763.994866
        Mode=randread : block_size=4K q_depth=1024 iops=2936709.315689
        Mode=randread : block_size=4K q_depth=2048 iops=2959925.452316
        Mode=randread : block_size=4K q_depth=4096 iops=2958421.601213
        Mode=randread : block_size=4K q_depth=8192 iops=2959127.796504
        Mode=randread : block_size=4K q_depth=16384 iops=2960391.000649
        Mode=randwrite : block_size=4K q_depth=64 iops=1130273.218037
        Mode=randwrite : block_size=4K q_depth=128 iops=1912386.740854
        Mode=randwrite : block_size=4K q_depth=256 iops=2318437.929198
        Mode=randwrite : block_size=4K q_depth=512 iops=2470136.201173
        Mode=randwrite : block_size=4K q_depth=1024 iops=2483698.263425
        Mode=randwrite : block_size=4K q_depth=2048 iops=2205233.195319
        Mode=randwrite : block_size=4K q_depth=4096 iops=2281327.683061
        Mode=randwrite : block_size=4K q_depth=8192 iops=2225102.752869
        Mode=randwrite : block_size=4K q_depth=16384 iops=2438168.399643
        Mode=read : block_size=4K q_depth=64 iops=828364.867922
        Mode=read : block_size=4K q_depth=128 iops=1428920.655052
        Mode=read : block_size=4K q_depth=256 iops=2250480.313791
        Mode=read : block_size=4K q_depth=512 iops=2867088.486120
        Mode=read : block_size=4K q_depth=1024 iops=2909384.708435
        Mode=read : block_size=4K q_depth=2048 iops=2933113.991344
        Mode=read : block_size=4K q_depth=4096 iops=2940695.149262
        Mode=read : block_size=4K q_depth=8192 iops=2937323.604771
        Mode=read : block_size=4K q_depth=16384 iops=2936471.737867
        Mode=write : block_size=4K q_depth=64 iops=1112600.639133
        Mode=write : block_size=4K q_depth=128 iops=1935590.568604
        Mode=write : block_size=4K q_depth=256 iops=2514642.866004
        Mode=write : block_size=4K q_depth=512 iops=2363649.028751
        Mode=write : block_size=4K q_depth=1024 iops=2365679.603555
        Mode=write : block_size=4K q_depth=2048 iops=2235795.103716
        Mode=write : block_size=4K q_depth=4096 iops=2378082.430258
        Mode=write : block_size=4K q_depth=8192 iops=2230193.164671
        Mode=write : block_size=4K q_depth=16384 iops=2462835.101398


Logs can be found at C:\LISAv2\GO85\TestResults\2019-05-11-18-22-25-1489

Also tested with other values. 
FIO_STRESS_RUNTIME=60: Execution time is 50 min
FIO_STRESS_RUNTIME=120: Execution time is 186 min

LiliDeng commented:
@ader1990 Do you know why we still see SC2086 error even we didn't include it?

LiliDeng commented:
Please fix remain errors. @v-sijliu

lizzha commented on Testscripts/Linux/NET-Copy-Both-Ways.sh:
```diff
@@ -145,6 +145,7 @@ remote_home=$(ssh -i "$HOME"/.ssh/"$SSH_PRIVATE_KEY" -o StrictHostKeyChecking=no
     exit 0
     ")
 sts=$?
+LogMsg "$remote_home"
```
if the variable is not used, can we remove the "remote_home="  and run the command on the right side directly?

lizzha commented on Testscripts/Linux/NET-Test-Network.sh:
```diff
@@ -189,6 +189,7 @@ if [ "${MAC:-UNDEFINED}" != "UNDEFINED" ]; then
     else
         LogMsg "MAC Address $MAC was found on the VM"   
     fi
+    LogMsg "$__sys_interface"
```
ditto

lizzha commented on Testscripts/Linux/STOR-Lis-Disk.sh:
```diff
@@ -82,6 +82,7 @@ LogMsg "constants disk count = $disk_count"
 for drive in /dev/sd*[^0-9]; do
+    LogMsg "$drive"
     sd_count=$((sd_count + 1))
 done
```
can use command "ls /dev/sd*[^0-9] | wc -l" to count the number. Thus the variable "drive" is not needed

lizzha commented on Testscripts/Linux/TestRDMA_MultiVM.sh:
```diff
@@ -90,6 +89,7 @@ for vm in $master $slaves_array; do
 	LogMsg "Checking $ib_nic status in $vm"
 	temp=$(ssh root@${vm} "ifconfig $ib_nic | grep 'inet '")
 	ib_nic_status=$?
+	LogMsg "$temp"
```
Run it directly? ssh root@${vm} "ifconfig $ib_nic | grep 'inet '"

lizzha commented on Testscripts/Linux/customKernelInstall.sh:
```diff
@@ -63,6 +63,7 @@ CheckInstallLockUbuntu()
     else
         LogMsg "No lock on dpkg present."
     fi
+    LogMsg "$dpkgPID"
```
ditto

lizzha commented on Testscripts/Linux/perf_fio_nfs.sh:
```diff
@@ -165,6 +163,7 @@ if [ $? -eq 0 ]; then
 	else
 		mdVolume="/dev/md0"
 	fi
+	LogMsg "$mdVolume"
```
please check is this variable used anywhere?

lizzha commented on Testscripts/Linux/utils.sh:
```diff
@@ -337,7 +337,7 @@ GetDistro()
 		;;
 	esac
-
+	echo "$OS_FAMILY"
```
Print more messages, "OS family: $OS_FAMILY"

lizzha commented on Testscripts/Linux/utils.sh:
```diff
@@ -2089,6 +2090,7 @@ GetGuestGeneration()
     else
         os_GENERATION=1
     fi
+	echo "$os_GENERATION"
```
Generation: $os_GENERATION

lizzha commented on Testscripts/Linux/utils.sh:
```diff
@@ -3284,6 +3286,7 @@ function test_rsync_files() {
 function change_mtu_increment() {
     test_iface=$1
     ignore_iface=$2
+    echo "$ignore_iface"
```
it seems a bug in this function, the variable used later is $iface_ignore, please fix it.

ader1990 commented:
@v-sijliu please add in the azure-pipelines.yml the shellcheck stage and rules that you are fixing. you can get the stage from the https://github.com/LIS/LISAv2/pull/447

ader1990 commented on Testscripts/Linux/BVT-NET-IFUP-IFDOWN.sh:
```diff
@@ -61,7 +61,7 @@ ReloadNetvsc()
 # Source constants file and initialize most common variables
```
@v-sijliu please add in the azure-pipelines.yml the shellcheck stage and rules that you are fixing. you can get the stage from the #447

LiliDeng replied:
@ader1990 already change azure-pipelines.yml  in this PR

LiliDeng commented on Testscripts/Linux/NET-Max-NICs.sh:
```diff
@@ -64,9 +64,6 @@ if [ "${TEST_TYPE:-UNDEFINED}" = "UNDEFINED" ]; then
     exit 0
-else
-    IFS=',' read -a TYPE <<< "$TEST_TYPE"
-fi
```
keep fi

LiliDeng commented:
For nested ones, pick up one case is enough, because only update the same command

  | Azure | Result | HyperV
-- | -- | -- | --
//Testscripts/Linux/BVT-NET-IFUP-IFDOWN.sh   (Edit) | BVT-NET-IFUP-IFDOWN | PASS | BVT-NET-IFUP-IFDOWN
//Testscripts/Linux/CollectLogFile.sh   (Edit) |   |   |  
//Testscripts/Linux/ConfigureSRIOV.sh   (Edit) | VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM | PASS |  
//Testscripts/Linux/ConfigureSRIOV-workaround.sh   (Edit) |   |   |  
//Testscripts/Linux/CORE-LogCollector.sh   (Edit) |   |   |  
//Testscripts/Linux/CreateRaid.sh   (Edit) | PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K | PASS |  
//Testscripts/Linux/customKernelInstall.sh   (Edit) | -CustomKernel "http://linuxpipelineshare.blob.core.windows.net/linuxpackages/linuxnext-kernel-latest.deb | PASS |  
//Testscripts/Linux/customLISInstall.sh   (Edit) |   |   |  
//Testscripts/Linux/fio_jason_parser.sh (Edit) （7）（10）（21）（23） | PERF-STORAGE-4K-IO | PASS | HYPERV-WINDOWS-NESTED-HYPERV-STORAGE-SINGLE-DISK
//Testscripts/Linux/InstallCUDADrivers.sh   (Edit) | VERIFY-NVIDIA-CUDA-DRIVER-FOR-NC6-VM   canonical   ubuntuserver 16.04-lts Latest | PASS |  
//Testscripts/Linux/LIS-Mount-FloppyDisk.sh   (Edit) |   |   | LIS-MOUNT-FLOPPYDISK
//Testscripts/Linux/nested_hyperv_ntttcp_different_l1_nat.sh   (Edit) |   |   | NESTED-HYPERV-NTTTCP-DIFFERENT-L1-NAT
//Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-HYPERV-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE
//Testscripts/Linux/nested_kvm_lagscope_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-KVM-LAGSCOPE-DIFFERENT-L1-PUBLIC-BRIDGE
//Testscripts/Linux/nested_kvm_netperf_pps.sh   (Edit) |   |   | NESTED-KVM-NETPERF-PPS
//Testscripts/Linux/nested_kvm_netperf_pps_nat.sh   (Edit) | AZURE-NESTED-KVM-NETPERF-PPS | Pass |  
//Testscripts/Linux/nested_kvm_ntttcp_different_l1_nat.sh   (Edit) | AZURE-NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT |   | NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT
//Testscripts/Linux/nested_kvm_ntttcp_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-KVM-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE
//Testscripts/Linux/nested_kvm_ntttcp_private_bridge.sh   (Edit) | NESTED-KVM-NTTTCP-PRIVATE-BRIDGE | Pass |  
//Testscripts/Linux/nested_kvm_perf_fio.sh   (Edit) |   |   |  
//Testscripts/Linux/nested_kvm_storage_perf.sh   (Edit) | AZURE-NESTED-KVM-STORAGE-SINGLE-DISK |   | HYPERV-NESTED-KVM-STORAGE-SINGLE-DISK
//Testscripts/Linux/nested_vm_utils.sh (Edit)    （13） |   |   |  
//Testscripts/Linux/NET-Configure-Vxlan.sh   (Edit) |   |   |  
//Testscripts/Linux/NET-Copy-Both-Ways.sh   (Edit) |   |   | NET-COPY-BOTH-WAYS
//Testscripts/Linux/NET-Max-NICs.sh   (Edit) |   |   | NET-MAX-SYNTHETIC-NICS   add fi
//Testscripts/Linux/NET-Netperf-Server.sh   (Edit) |   |   | ETHTOOL-CHECK-STATISTICS
//Testscripts/Linux/NET-Test-Legacy.sh   (Edit) |   |   | NET-LEGACY
//Testscripts/Linux/NET-Test-Network.sh   (Edit) |   |   | NET-EXTERNAL
//Testscripts/Linux/perf_fio.sh   (Edit) | PERF-STORAGE-4K-IO | PASS | PERF-STORAGE-4K-IO
//Testscripts/Linux/perf_fio_nfs.sh (Edit) | PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K |   |  
//Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh   (Edit) |   |   | VHDX-RESIZE-TO-BIGGER512
//Testscripts/Linux/STOR_VHDXResize_PartitionDiskOver2TB.sh   (Edit) |   |   | VHDX-RESIZE-OVER2TB-SCSI
//Testscripts/Linux/STOR_VSS_Disk_Stress.sh   (Edit) |   |   | STOR-VSS-BACKUP-RESTORE-DISKSTRESS
//Testscripts/Linux/STOR_VSS_ISCSI_PartitionDisks.sh   (Edit) |   |   | PRODUCTION-CHECKPOINT-iSCSI
//Testscripts/Linux/STOR_VSS_Set_VSS_Daemon.sh   (Edit) |   |   | STOR-VSS-BACKUP-CHANGE-HYPERVVSSD
//Testscripts/Linux/STOR_VSS_StopNetwork.sh   (Edit) |   |   | PRODUCTION-CHECKPOINT-ISO-NONETWORK
//Testscripts/Linux/STOR-Lis-Disk.sh   (Edit) |   |   | STORAGE-VHDX-IDE-FIXED
//Testscripts/Linux/Sysbench.sh   (Edit) |   |   | STRESSTEST-SYSBENCH
//Testscripts/Linux/TestRDMA_MultiVM.sh   (Edit) | VERIFY-INFINIBAND-INTEL-MPI-2VM-STANDARD-A8 | Aborted |  
//Testscripts/Linux/verify-ifdown-ifup-eth0-rmmod-modprobe-hv-netvsc.sh   (Edit) | VERIFY-IFUP-IFDOWN-RELOAD-HV_NETVSC-Synthetic   canonical   ubuntuserver 16.04-lts Latest | Pass |  


  | Azure | Result | HyperV | Result
-- | -- | -- | -- | --
//Testscripts/Linux/BVT-NET-IFUP-IFDOWN.sh   (Edit) | BVT-NET-IFUP-IFDOWN | PASS | BVT-NET-IFUP-IFDOWN |  
//Testscripts/Linux/CollectLogFile.sh   (Edit) |   |   |   |  
//Testscripts/Linux/ConfigureSRIOV.sh   (Edit) | VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM | PASS |   |  
//Testscripts/Linux/ConfigureSRIOV-workaround.sh   (Edit) |   |   |   |  
//Testscripts/Linux/CORE-LogCollector.sh   (Edit) |   |   |   |  
//Testscripts/Linux/CreateRaid.sh   (Edit) | PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K | PASS |   |  
//Testscripts/Linux/customKernelInstall.sh   (Edit) | -CustomKernel "http://linuxpipelineshare.blob.core.windows.net/linuxpackages/linuxnext-kernel-latest.deb | PASS |   |  
//Testscripts/Linux/customLISInstall.sh   (Edit) |   |   |   |  
//Testscripts/Linux/fio_jason_parser.sh (Edit) （7）（10）（21）（23） | PERF-STORAGE-4K-IO | PASS | HYPERV-WINDOWS-NESTED-HYPERV-STORAGE-SINGLE-DISK |  
//Testscripts/Linux/InstallCUDADrivers.sh   (Edit) | VERIFY-NVIDIA-CUDA-DRIVER-FOR-NC6-VM   canonical   ubuntuserver 16.04-lts Latest | PASS |   |  
//Testscripts/Linux/LIS-Mount-FloppyDisk.sh   (Edit) |   |   | LIS-MOUNT-FLOPPYDISK | PASS
//Testscripts/Linux/nested_hyperv_ntttcp_different_l1_nat.sh   (Edit) |   |   | NESTED-HYPERV-NTTTCP-DIFFERENT-L1-NAT |  
//Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-HYPERV-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE |  
//Testscripts/Linux/nested_kvm_lagscope_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-KVM-LAGSCOPE-DIFFERENT-L1-PUBLIC-BRIDGE |  
//Testscripts/Linux/nested_kvm_netperf_pps.sh   (Edit) |   |   | NESTED-KVM-NETPERF-PPS |  
//Testscripts/Linux/nested_kvm_netperf_pps_nat.sh   (Edit) | AZURE-NESTED-KVM-NETPERF-PPS | Pass |   |  
//Testscripts/Linux/nested_kvm_ntttcp_different_l1_nat.sh   (Edit) | AZURE-NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT |   | NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT |  
//Testscripts/Linux/nested_kvm_ntttcp_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-KVM-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE |  
//Testscripts/Linux/nested_kvm_ntttcp_private_bridge.sh   (Edit) | NESTED-KVM-NTTTCP-PRIVATE-BRIDGE | Pass |   |  
//Testscripts/Linux/nested_kvm_perf_fio.sh   (Edit) |   |   |   |  
//Testscripts/Linux/nested_kvm_storage_perf.sh   (Edit) | AZURE-NESTED-KVM-STORAGE-SINGLE-DISK |   | HYPERV-NESTED-KVM-STORAGE-SINGLE-DISK |  
//Testscripts/Linux/nested_vm_utils.sh (Edit)    （13） |   |   |   |  
//Testscripts/Linux/NET-Configure-Vxlan.sh   (Edit) |   |   |   |  
//Testscripts/Linux/NET-Copy-Both-Ways.sh   (Edit) |   |   | NET-COPY-BOTH-WAYS | PASS
//Testscripts/Linux/NET-Max-NICs.sh   (Edit) |   |   | NET-MAX-SYNTHETIC-NICS   add fi | PASS
//Testscripts/Linux/NET-Netperf-Server.sh   (Edit) |   |   | ETHTOOL-CHECK-STATISTICS | Fail   same with master
//Testscripts/Linux/NET-Test-Legacy.sh   (Edit) |   |   | NET-LEGACY | PASS
//Testscripts/Linux/NET-Test-Network.sh   (Edit) |   |   | NET-EXTERNAL | PASS
//Testscripts/Linux/perf_fio.sh   (Edit) | PERF-STORAGE-4K-IO | PASS | PERF-STORAGE-4K-IO |  
//Testscripts/Linux/perf_fio_nfs.sh (Edit) | PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K |   |   |  
//Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh   (Edit) |   |   | VHDX-RESIZE-TO-BIGGER512 | PASS
//Testscripts/Linux/STOR_VHDXResize_PartitionDiskOver2TB.sh   (Edit) |   |   | VHDX-RESIZE-OVER2TB-SCSI | PASS
//Testscripts/Linux/STOR_VSS_Disk_Stress.sh   (Edit) |   |   | STOR-VSS-BACKUP-RESTORE-DISKSTRESS |  
//Testscripts/Linux/STOR_VSS_ISCSI_PartitionDisks.sh   (Edit) |   |   | PRODUCTION-CHECKPOINT-iSCSI |  
//Testscripts/Linux/STOR_VSS_Set_VSS_Daemon.sh   (Edit) |   |   | STOR-VSS-BACKUP-CHANGE-HYPERVVSSD | PASS
//Testscripts/Linux/STOR_VSS_StopNetwork.sh   (Edit) |   |   | PRODUCTION-CHECKPOINT-ISO-NONETWORK | PASS
//Testscripts/Linux/STOR-Lis-Disk.sh   (Edit) |   |   | STORAGE-VHDX-IDE-FIXED | PASS
//Testscripts/Linux/Sysbench.sh   (Edit) |   |   | STRESSTEST-SYSBENCH | PASS
//Testscripts/Linux/TestRDMA_MultiVM.sh   (Edit) | VERIFY-INFINIBAND-INTEL-MPI-2VM-STANDARD-A8 | Aborted |   |  
//Testscripts/Linux/verify-ifdown-ifup-eth0-rmmod-modprobe-hv-netvsc.sh   (Edit) | VERIFY-IFUP-IFDOWN-RELOAD-HV_NETVSC-Synthetic   canonical   ubuntuserver 16.04-lts Latest | Pass |   |  

LiliDeng commented:
For nested ones, pick up one case is enough, because only update the same command


  | Azure | Result | HyperV
-- | -- | -- | --
//Testscripts/Linux/BVT-NET-IFUP-IFDOWN.sh   (Edit) | BVT-NET-IFUP-IFDOWN | PASS | BVT-NET-IFUP-IFDOWN
//Testscripts/Linux/CollectLogFile.sh   (Edit) |   |   |  
//Testscripts/Linux/ConfigureSRIOV.sh   (Edit) | VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM | PASS |  
//Testscripts/Linux/ConfigureSRIOV-workaround.sh   (Edit) |   |   |  
//Testscripts/Linux/CORE-LogCollector.sh   (Edit) |   |   |  
//Testscripts/Linux/CreateRaid.sh   (Edit) | PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K | PASS |  
//Testscripts/Linux/customKernelInstall.sh   (Edit) | -CustomKernel "http://linuxpipelineshare.blob.core.windows.net/linuxpackages/linuxnext-kernel-latest.deb | PASS |  
//Testscripts/Linux/customLISInstall.sh   (Edit) |   |   |  
//Testscripts/Linux/fio_jason_parser.sh (Edit) （7）（10）（21）（23） | PERF-STORAGE-4K-IO | PASS | HYPERV-WINDOWS-NESTED-HYPERV-STORAGE-SINGLE-DISK
//Testscripts/Linux/InstallCUDADrivers.sh   (Edit) | VERIFY-NVIDIA-CUDA-DRIVER-FOR-NC6-VM   canonical   ubuntuserver 16.04-lts Latest | PASS |  
//Testscripts/Linux/LIS-Mount-FloppyDisk.sh   (Edit) |   |   | LIS-MOUNT-FLOPPYDISK
//Testscripts/Linux/nested_hyperv_ntttcp_different_l1_nat.sh   (Edit) |   |   | NESTED-HYPERV-NTTTCP-DIFFERENT-L1-NAT
//Testscripts/Linux/nested_hyperv_ntttcp_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-HYPERV-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE
//Testscripts/Linux/nested_kvm_lagscope_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-KVM-LAGSCOPE-DIFFERENT-L1-PUBLIC-BRIDGE
//Testscripts/Linux/nested_kvm_netperf_pps.sh   (Edit) |   |   | NESTED-KVM-NETPERF-PPS
//Testscripts/Linux/nested_kvm_netperf_pps_nat.sh   (Edit) | AZURE-NESTED-KVM-NETPERF-PPS | Pass |  
//Testscripts/Linux/nested_kvm_ntttcp_different_l1_nat.sh   (Edit) | AZURE-NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT |   | NESTED-KVM-NTTTCP-DIFFERENT-L1-NAT
//Testscripts/Linux/nested_kvm_ntttcp_different_l1_public_bridge.sh   (Edit) |   |   | NESTED-KVM-NTTTCP-DIFFERENT-L1-PUBLIC-BRIDGE
//Testscripts/Linux/nested_kvm_ntttcp_private_bridge.sh   (Edit) | NESTED-KVM-NTTTCP-PRIVATE-BRIDGE | Pass |  
//Testscripts/Linux/nested_kvm_perf_fio.sh   (Edit) |   |   |  
//Testscripts/Linux/nested_kvm_storage_perf.sh   (Edit) | AZURE-NESTED-KVM-STORAGE-SINGLE-DISK |   | HYPERV-NESTED-KVM-STORAGE-SINGLE-DISK
//Testscripts/Linux/nested_vm_utils.sh (Edit)    （13） |   |   |  
//Testscripts/Linux/NET-Configure-Vxlan.sh   (Edit) |   |   |  
//Testscripts/Linux/NET-Copy-Both-Ways.sh   (Edit) |   |   | NET-COPY-BOTH-WAYS
//Testscripts/Linux/NET-Max-NICs.sh   (Edit) |   |   | NET-MAX-SYNTHETIC-NICS   add fi
//Testscripts/Linux/NET-Netperf-Server.sh   (Edit) |   |   | ETHTOOL-CHECK-STATISTICS
//Testscripts/Linux/NET-Test-Legacy.sh   (Edit) |   |   | NET-LEGACY
//Testscripts/Linux/NET-Test-Network.sh   (Edit) |   |   | NET-EXTERNAL
//Testscripts/Linux/perf_fio.sh   (Edit) | PERF-STORAGE-4K-IO | PASS | PERF-STORAGE-4K-IO
//Testscripts/Linux/perf_fio_nfs.sh (Edit) | PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K |   |  
//Testscripts/Linux/STOR_VHDXResize_PartitionDisk.sh   (Edit) |   |   | VHDX-RESIZE-TO-BIGGER512
//Testscripts/Linux/STOR_VHDXResize_PartitionDiskOver2TB.sh   (Edit) |   |   | VHDX-RESIZE-OVER2TB-SCSI
//Testscripts/Linux/STOR_VSS_Disk_Stress.sh   (Edit) |   |   | STOR-VSS-BACKUP-RESTORE-DISKSTRESS
//Testscripts/Linux/STOR_VSS_ISCSI_PartitionDisks.sh   (Edit) |   |   | PRODUCTION-CHECKPOINT-iSCSI
//Testscripts/Linux/STOR_VSS_Set_VSS_Daemon.sh   (Edit) |   |   | STOR-VSS-BACKUP-CHANGE-HYPERVVSSD
//Testscripts/Linux/STOR_VSS_StopNetwork.sh   (Edit) |   |   | PRODUCTION-CHECKPOINT-ISO-NONETWORK
//Testscripts/Linux/STOR-Lis-Disk.sh   (Edit) |   |   | STORAGE-VHDX-IDE-FIXED
//Testscripts/Linux/Sysbench.sh   (Edit) |   |   | STRESSTEST-SYSBENCH
//Testscripts/Linux/TestRDMA_MultiVM.sh   (Edit) | VERIFY-INFINIBAND-INTEL-MPI-2VM-STANDARD-A8 | Aborted |  
//Testscripts/Linux/verify-ifdown-ifup-eth0-rmmod-modprobe-hv-netvsc.sh   (Edit) | VERIFY-IFUP-IFDOWN-RELOAD-HV_NETVSC-Synthetic   canonical   ubuntuserver 16.04-lts Latest | Pass |  

LiliDeng commented:
@lizzha Done for testing.

BVT-NET-IFUP-IFDOWN.sh
BVT-NET-IFUP-IFDOWN PASS

ConfigureSRIOV.sh
VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM PASS

CreateRaid.sh
PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K PASS

customKernelInstall.sh
-CustomKernel "http://linuxpipelineshare.blob.core.windows.net/linuxpackages/linuxnext-kernel-latest.deb PASS

customLISInstall.sh
-CustomLIS lisnext PASS

fio_jason_parser.sh
PERF-STORAGE-4K-IO PASS

InstallCUDADrivers.sh
VERIFY-NVIDIA-CUDA-DRIVER-FOR-NC6-VM PASS

LIS-Mount-FloppyDisk.sh
LIS-MOUNT-FLOPPYDISK PASS

nested_kvm_netperf_pps.sh
NESTED-KVM-NETPERF-PPS Pass

nested_kvm_netperf_pps_nat.sh
AZURE-NESTED-KVM-NETPERF-PPS Pass

nested_kvm_ntttcp_private_bridge.sh
NESTED-KVM-NTTTCP-PRIVATE-BRIDGE Pass

NET-Configure-Vxlan.sh
NET-Copy-Both-Ways.sh
NET-COPY-BOTH-WAYS Pass

NET-Max-NICs.sh 
NET-MAX-SYNTHETIC-NICS Pass

NET-Netperf-Server.sh
ETHTOOL-CHECK-STATISTICS Fail, same with master

NET-Test-Legacy.sh
NET-LEGACY Pass

NET-Test-Network.sh
NET-EXTERNAL Pass

perf_fio.sh
PERF-STORAGE-4K-IO PASS

perf_fio_nfs.sh
PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K PASS

STOR_VHDXResize_PartitionDisk.sh
VHDX-RESIZE-TO-BIGGER512 PASS

STOR_VHDXResize_PartitionDiskOver2TB.sh
VHDX-RESIZE-OVER2TB-SCSI PASS

STOR_VSS_Disk_Stress.sh
STOR-VSS-BACKUP-RESTORE-DISKSTRESS Fail, same with master

STOR_VSS_ISCSI_PartitionDisks.sh
PRODUCTION-CHECKPOINT-iSCSI Fail, same with master
 
STOR_VSS_Set_VSS_Daemon.sh
STOR-VSS-BACKUP-CHANGE-HYPERVVSSD PASS

STOR_VSS_StopNetwork.sh
PRODUCTION-CHECKPOINT-ISO-NONETWORK PASS

STOR-Lis-Disk.sh
STORAGE-VHDX-IDE-FIXED PASS

Sysbench.sh
STRESSTEST-SYSBENCH PASS

TestRDMA_MultiVM.sh
VERIFY-INFINIBAND-INTEL-MPI-2VM-STANDARD-A8
Same result with master, download log timeout

verify-ifdown-ifup-eth0-rmmod-modprobe-hv-netvsc.sh
VERIFY-IFUP-IFDOWN-RELOAD-HV_NETVSC-Synthetic Pass

lizzha commented on Testscripts/Linux/utils.sh:
```diff
@@ -686,6 +686,7 @@ NetmaskToCidr()
 
 	declare -i netbits=0
 	oldifs="$IFS"
+	echo "$oldifs"
```
this variable is not used, can be removed

juhlee-microsoft commented:
@LiliDeng are we still working on 4 errors? @adriansuhov fixed the log downloading timeout in setupRDMA branch before. Once it is merging to master, below problem will be resolved. 


_**TestRDMA_MultiVM.sh
VERIFY-INFINIBAND-INTEL-MPI-2VM-STANDARD-A8
Same result with master, download log timeout**_

Need to clarify what we are going to do with remaining 4 failed cases. @lizzha already signed off.

LiliDeng commented:
@juhlee-microsoft You mean the conflict? We will fix it.

LiliDeng commented:
@juhlee-microsoft We are sure that those failures not cause by our change.

lizzha commented:
@juhlee-microsoft for the failed cases, if they're test issues, Lili's team will fix them. They're cleaning up test issues on Azure case by case now.
When do you plan to merge the setupRDMA branch to master?

juhlee-microsoft commented:
Plan it around beginning of Feb.

________________________________
From: Liz Zhang <notifications@github.com>
Sent: Thursday, January 24, 2019 8:40:56 PM
To: LIS/LISAv2
Cc: Ju-Hyoung Lee; Mention
Subject: Re: [LIS/LISAv2] shellcheck: fix SC2006,SC2034,SC2242,SC2241,SC2239 warnings (#629)


@juhlee-microsoft<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjuhlee-microsoft&data=02%7C01%7Cjuhlee%40microsoft.com%7Cca95ae01c100427f2e1b08d6827f4bb2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636839880623669351&sdata=D5cHIU25skHUzjGfeyTibaHiE6jd4OmukhDmBDii6f0%3D&reserved=0> for the failed cases, if they're test issues, Lili's team will fix them. They're cleaning up test issues on Azure case by case now.
When do you plan to merge the setupRDMA branch to master?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FLIS%2FLISAv2%2Fpull%2F629%23issuecomment-457453635&data=02%7C01%7Cjuhlee%40microsoft.com%7Cca95ae01c100427f2e1b08d6827f4bb2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636839880623669351&sdata=8PIR0oNXDfliVC46pY0mZr5CJfAmBbPc8eCtlCgFdoU%3D&reserved=0>, or mute the thread<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAWAC18P4V_X23FMYyT8SAggiN82yBdJLks5vGorYgaJpZM4aKBau&data=02%7C01%7Cjuhlee%40microsoft.com%7Cca95ae01c100427f2e1b08d6827f4bb2%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636839880623679359&sdata=1GgtBciUFtqawWCme0hGJEbLaihZL%2Bls60%2BZPtijJ9o%3D&reserved=0>.

LiliDeng commented:
Pick up 2 cases to run against latest code, passed, this PR is ready for merged @lizzha Thanks!

juhlee-microsoft committed:
NVMe stress initial checkin

1. Added a new stress test based on Perf 4K IO test script
2. Added a new runtime value in the xml file.

juhlee-microsoft committed:
Change to 12h for 4 queues

juhlee-microsoft committed:
Change maxIO to 1

And added a few logging msg

juhlee-microsoft committed:
Removed spaces

juhlee-microsoft committed:
Put back to maxIO 4

juhlee-microsoft committed:
Merge branch 'juhlee/nvme-io-stress' of https://github.com/LIS/LISAv2 into juhlee/nvme-io-stress

juhlee-microsoft committed:
Set the default value to 7 hr 30 min testing time.

juhlee-microsoft committed:
Merge branch 'master' into juhlee/nvme-io-stress
